packages/swingset-liveslots/src/collectionManager.js

```diff
@@ -647,48 +682,39 @@ export function makeCollectionManager(
     const { id, subid } = parseVatSlot(vobjID);
     const kindName = storeKindIDToName.get(`${id}`);
     assert(kindName, `unknown kind ID ${id}`);
-    const collection = summonCollectionInternal(false, 'test', subid, kindName);
+    const collectionID = `${subid}`;
+    const collection = summonCollectionInternal(false, collectionID, kindName);
     return collection.sizeInternal();
   }
 
   function deleteCollection(vobjID) {
-    if (!allCollectionObjIDs.has(vobjID)) {
-      return false; // already deleted
-    }
     const { id, subid } = parseVatSlot(vobjID);
     const kindName = storeKindIDToName.get(`${id}`);
-    const collection = summonCollectionInternal(false, 'GC', subid, kindName);
-    allCollectionObjIDs.delete(vobjID);
+    const collectionID = `${subid}`;
+    const collection = summonCollectionInternal(false, collectionID, kindName);
 
     const doMoreGC = collection.clearInternal(true);
-    for (const dbKey of enumerateKeysWithPrefix(syscall, prefixc(subid, '|'))) {
+    for (const dbKey of enumerateKeysWithPrefix(
+      syscall,
+      prefixc(collectionID, '|'),
+    )) {
+      if (dbKey.endsWith('|schemata') || dbKey.endsWith('|label')) {
+        continue; // will be deleted when schemaCache is flushed
+      }
```

Comment:
I think I understand why you're doing this, but it still seems kinda weird and off.

Reply:
Yeah, true. The `schemaCache` owns those two keys, but there are others (`|nextOrdinal` and `|entryCount`), and they need to be deleted too.

I wanted to keep the PR minimal, but one option that might be cleaner would be to have `schemaCache` own *all* the metadata keys (and maybe call it something else, `collectionMetadataCache`?). We could put all the values in a single DB key, and have the cache read them all. In that approach, we'd write to the cache more often (any time `init` or `set` introduced a new ordinal/Remotable, or any time `init` or `delete` changed the `entryCount`), and at the end of each of those cranks, we'd write out all of them. But it seemed better to separate the "hot" keys (`|nextOrdinal` and `|entryCount`) from the "cold" / immutable-after-creation ones.

I'll add a bit more explanation to the comment.